src/base/bittorrent/private/nativesessionextension.cpp

```diff
@@ -60,7 +60,7 @@ std::shared_ptr<lt::torrent_plugin> NativeSessionExtension::new_torrent(const lt
 #else
 boost::shared_ptr<lt::torrent_plugin> NativeSessionExtension::new_torrent(const lt::torrent_handle &torrentHandle, void *)
 {
-    return boost::shared_ptr<lt::torrent_plugin> {new NativeTorrentExtension {torrentHandle}};
+    return boost::make_shared<NativeTorrentExtension>(torrentHandle);
```

Comment:
It didn't compile for me when I implemented this code, so I had to go back to using the shared_ptr constructor instead.

Reply:
Our CI builds didn't error out, maybe you will want to try it again.
Also this function already exists in our minimum boost version requirement: 1.40
https://www.boost.org/doc/libs/1_40_0/libs/smart_ptr/make_shared.html

Reply:
...or maybe it was due to similar code from "bittorrent/session.cpp"... (isn't affected by this PR for some reason)
>Also this function already exists in our minimum boost version requirement: 1.40

The error was due to different types of shared_ptr (passing shared_ptr for derived class as shared_ptr for base class).

Reply:
>The error was due to different types of shared_ptr (passing shared_ptr for derived class as shared_ptr for base class).

Hope that doesn't mean you still experience the issue, otherwise I'll have to revert it.

Reply:
Maybe some older boost versions have no such conversions... I can test it again with my current development environment. But we really need to test it with minimal supported boost version.

Reply:
>...or maybe it was due to similar code from "bittorrent/session.cpp"... (isn't affected by this PR for some reason)

https://github.com/qbittorrent/qBittorrent/blob/ca01b049a600a65a14cc0b0fcf1c13817b1c1f4e/src/base/bittorrent/session.cpp#L1133
So why didn't you change it too?

Reply:
>So why didn't you change it too?

I pushed an update and CI fails now: https://ci.appveyor.com/project/qbittorrent/qbittorrent/builds/31261773#L405
Trying some workarounds.

Reply:
Seems I need to write:
```c++
const boost::shared_ptr<lt::plugin> sessionExt = boost::make_shared<NativeSessionExtension>();
m_nativeSession->add_extension(sessionExt);
```
Which is even longer than the original, I think it is not worth it so I'm going to leave it alone.